### PR TITLE
Compile examples in CI

### PR DESCRIPTION
**Pull request by ahcorde**
Related with this issue https://github.com/ignitionrobotics/ign-physics/issues/109

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by chapulina** [comment]
Thanks, this solution works, but it doesn't scale well. When new examples are created, we may forget to add them to CI. This is also only running on actions, but not on Jenkins for Mac for exmple.

Did you look into what it would take to do the same as other libraries like `ign-math` for example?

https://github.com/ignitionrobotics/ign-math/blob/master/test/integration/ExamplesBuild_TEST.cc

**Comment by ahcorde** [comment]
@chapulina added a integration test to compile examples https://github.com/ignitionrobotics/ign-physics/pull/133/commits/b0044911a18381842cbc74aade34639c98b829a9

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -6,17 +6,43 @@ include_directories (
...
 set(GTEST_MAIN_LIBRARY "${PROJECT_BINARY_DIR}/test/libgtest_main.a")
 
+execute_process(COMMAND cmake -E remove_directory ${CMAKE_BINARY_DIR}/test_results)
+execute_process(COMMAND cmake -E make_directory ${CMAKE_BINARY_DIR}/test_results)
 # This gives us access to src/utils headers
-include_directories(${PROJECT_SOURCE_DIR}/src)
+include_directories(${PROJECT_SOURCE_DIR}/src ${GTEST_INCLUDE_DIRS})
+
+#============================================================================
+# Do a fake install of ign-math in order to test the examples.
```
Comment (chapulina) [inline test/CMakeLists.txt:27]:
```suggestion
# Do a fake install of ign-physics in order to test the examples.
```

File: test/integration/CMakeLists.txt [hunk 2]
```diff
@@ -5,6 +5,11 @@ if (NOT DART_FOUND)
   list(REMOVE_ITEM tests DoublePendulum.cc)
 endif()
 
+configure_file (test_config.h.in ${PROJECT_BINARY_DIR}/test_config.h)
+
+# set(THREADS_PREFER_PTHREAD_FLAG ON)
+# find_package(Threads REQUIRED)
```
Comment (chapulina) [inline test/integration/CMakeLists.txt:11]: Can we remove these comments?

File: examples/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
```
Comment (chapulina) [inline examples/CMakeLists.txt:1]: Each example already has its own `CMakeLists`. Would it be possible to remove this top-level `CMakeLists`, which has duplicate information, and instead, in the test, iterate over all the directories inside `examples`?

**Comment by ahcorde** [comment]
@osrf-jenkins retest this please

File: test/integration/ExamplesBuild_TEST.cc [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+  return true;
+}
+
+#else
+#include <io.h>  // NOLINT(build/include_order)
+#include <windows.h>  // NOLINT(build/include_order)
+#include <winioctl.h>  // NOLINT(build/include_order)
+#include <winnt.h>  // NOLINT(build/include_order)
+#include <cstdint>
+#include "./win_dirent.h"
```
Comment (chapulina) [inline test/integration/ExamplesBuild_TEST.cc:92]: Are you missing this file?

```
D:\Jenkins\workspace\ign_physics-pr-win\ws\ign-physics\test\integration\ExamplesBuild_TEST.cc(92,10): fatal error C1083: Cannot open include file: './win_dirent.h': No such file or directory [D:\Jenkins\workspace\ign_physics-pr-win\ws\build\ignition-physics3\test\integration\INTEGRATION_ExamplesBuild_TEST.vcxproj]
```

**Comment by ahcorde** [comment]
friendly ping @chapulina

**Review by chapulina — APPROVED** [review]
Nice! LGTM

**Pull request merged**